### PR TITLE
Add book

### PR DESCRIPTION
**Pull request by smc395**
Adds the "Add book" functionality for an admin

**Significant Changes**
1. Added creation of books table (see LocalDB and BuildScript repo)
2. Unit tests in Backend
3. Form to add new book in Frontend
4. Backend implementation to handle request and form model

**How to Test**
1. Login
2. Confirm dropdown in **Admin Portal** can select **Create New Book**
3. Go to admin page and confirm you can select between **Change User Role** and **Add New Book**
4. Fill out **Title** and **Description** and confirm **Create Book** is enabled
5. Add a new chapter and confirm **Create Book** is disabled
6. Fill out **Chapter Title** field and confirm **Create Book** is enabled
7. Repeat for adding/removing Chapters, Sections, or Problems
8. After pressing **Create Book** confirm book you created is in your LocalDB

**Notes**
:watermelon:

### CODE REVIEW

File: Backend/mentii/book_ctrl.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+  #g will be not be available during testing
+  #userRole will need to be passed to the function
+  if g:
+    userRole = g.authenticatedUser['userRole']
+  #role is confirmed here incase createBook is called from somewhere other
+  #than app.py createBook()
+  if userRole != 'admin':
+    response.addError('Role error', 'Only admins can create books')
+  elif bookData is None:
```
Comment (smc395) [inline Backend/mentii/book_ctrl.py]: Added check that title and description are in data. chapters field is not required as we do not make it required as per our discussion.

File: Backend/tests/test_book_ctrl.py [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+    userRole = 'admin'
+    response = book_ctrl.createBook(bookData, dynamodb, userRole)
+    # check response message
+    self.assertEqual(response.payload, {'Success': 'Book Created'})
+
+    # check number of books in table after creating
+    scan = self.booksTable.scan()
+    self.assertEqual(scan['Count'] , count+1)
+
+  def test_createBook_fail(self):
```
Comment (smc395) [inline Backend/tests/test_book_ctrl.py]: Added tests to check for valid book data

File: Backend/mentii/book_ctrl.py [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+  #g will be not be available during testing
+  #userRole will need to be passed to the function
+  if g:
+    userRole = g.authenticatedUser['userRole']
+  #role is confirmed here incase createBook is called from somewhere other
+  #than app.py createBook()
+  if userRole != 'admin':
+    response.addError('Role error', 'Only admins can create books')
+  elif 'title' not in bookData.keys() or 'description' not in bookData.keys():
```
Comment (RyanYoung25) [inline Backend/mentii/book_ctrl.py]: What about chapters? You use it on line 33.

File: Backend/tests/test_book_ctrl.py [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+    # check number of books in table before creating
+    scan = self.booksTable.scan()
+    count = scan['Count']
+
+    response = book_ctrl.createBook(bookData, dynamodb, userRole)
+    # check response message
+    error = response.errors[0]
+    self.assertEqual(error, {'message': 'Invalid book data given.', 'title': 'Book creation failed.'})
+
+    # check number of books in table stayed the same
```
Comment (RyanYoung25) [inline Backend/tests/test_book_ctrl.py]: Tests are better but you're still not testing if you didn't provide a chapter.

File: Backend/mentii/book_ctrl.py (hunk 3 above)
Comment (smc395) [inline Backend/mentii/book_ctrl.py]: Added bookData.get('chapters',[ ]) on line 33

File: Backend/tests/test_book_ctrl.py (hunk 4 above)
Comment (smc395) [inline Backend/tests/test_book_ctrl.py]: Added test for chapters

**Pull request merged**